src/lib/DepositController.js

```diff
@@ -54,47 +57,76 @@ export class DepositController {
     );
 
     let payload = recordSerializer.serialize(draft);
-
-    this.validate(payload);
     let response = {};
-    try {
-      if (!this.draftAlreadyCreated(payload)) {
-        response = await this.createDraft(payload, { store });
-      } else {
-        response = await this.apiClient.save(payload);
-      }
+    if (!this.draftAlreadyCreated(payload)) {
+      response = await this.createDraft(payload, { store });
+    } else {
+      response = await this.apiClient.save(payload);
+    }
+
+    let data = recordSerializer.deserialize(response.data || {});
+    let errors = recordSerializer.deserializeErrors(response.errors || []);
+
+    // response 100% successful
+    if ( 200 <= response.code && response.code < 300 && _isEmpty(errors) ) {
```

Comment:
Maybe extract this condition in the `DepositApiClientResponse` class and then you do something like `if response.is_ok()`. WDYT?